File: backend/src/main/java/vkr/backend/services/DialogService.java
package vkr.backend.services;

import vkr.backend.entities.Dialog;

import java.util.List;
import java.util.Optional;

public interface DialogService {
    /**
     * Получает диалог по id отправителя и получателя
     *
     * @param senderId id отправителя
     * @param recipientId id получателя
     * @return найденный диалог, если существует
     */
    Optional<Dialog> getDialog(Long senderId, Long recipientId);

    /**
     * Создает новый диалог
     *
     * @param senderId id отправителя
     * @param recipientId id получателя
     * @return созданный диалог
     */
    Dialog createDialog(Long senderId, Long recipientId);

    /**
     * Удаляет диалог по id
     *
     * @param id id диалога
     */
    void deleteDialog(Long id);

    /**
     * Ищет все диалоги для пользователя
     *
     * @param userId id пользователя
     * @return список диалогов
     */
    List<Dialog> findAllDialogsByUserId(Long userId);
}

File: frontend/src/services/dialogService.js
import axios from "axios";
import authHeader from "./authHeader";

const API_URL = "/api/dialogs";

const dialogService = {
    getDialog: (senderId, recipientId) => {
        return axios.get(`${API_URL}/find`, {
            params: {senderId, recipientId},
            headers: authHeader()
        }).then(response => {
            return response.data;
        });
    },

    createDialog: (senderId, recipientId) => {
        return axios.post(`${API_URL}/create`, null, {
            params: {senderId, recipientId},
            headers: authHeader()
        }).then(response => {
            return response.data;
        });
    },

    deleteDialog: (id) => {
        return axios.delete(`${API_URL}/delete/${id}`, {
            headers: authHeader()
        });
    },

    getAllDialogsByUserId: (userId) => {
        return axios.get(`${API_URL}/user/${userId}`, {
            headers: authHeader()
        }).then(response => {
            return response.data;
        });
    }
};

export default dialogService;
